src/nnet3/nnet-test-utils.cc

```diff
@@ -888,13 +888,21 @@ void ComputeExampleComputationRequestSimple(
   request->outputs.clear();
   inputs->clear();
 
+  int32 ivector_period = GetInputInterval(nnet, "ivector");
   std::vector<Index> input_indexes, ivector_indexes, output_indexes;
   for (int32 n = n_offset; n < n_offset + num_examples; n++) {
     for (int32 t = input_start_frame; t < input_end_frame; t++)
       input_indexes.push_back(Index(n, t, 0));
     for (int32 t = output_start_frame; t < output_end_frame; t++)
       output_indexes.push_back(Index(n, t, 0));
-    ivector_indexes.push_back(Index(n, 0, 0));
+    if (ivector_period == 0) // case 1: single ivector for the entire chunk
+      ivector_indexes.push_back(Index(n, 0, 0));
+    else { // case 2: multiple ivectors
+      int32 t_begin = std::floor(1.0 * input_start_frame / ivector_period);
+      int32 t_end = std::floor((input_end_frame - 1.0) / ivector_period);
```

Comment:
whenever 'end' appears in a variable name it implies it is one past the last one.  Here you should name your variables i_first and i_last if you want the last to be the last, not one past the last.  (don't call them t because that would imply they are time indexes).